### DISCUSSION TITLE
Force a device type that I know is working correctly.

### DISCUSSION THREAD

**Discussion by Alvanosa-code** [make-all/tuya-local#606] [Q&A] [answered]
I'm trying to add 4 fans with light control, and two of them show up with several devices where I could choose, after chosing the ledkia_fan_light device everything is working correctly.

For the other two fans that are exactly the same one, there is only one incorrect device type showing up, to be precise it is rgbcw_lightbulb. Is there any way for me to force it to be a ledkia_fan_light device?

Thanks in advance.

**Comment by make-all** [comment, marked as answer]
Although your devices seem superficially the same, the firmware apparently is different if they are returning different data, so there is no guarantee of compatibility.

What you can do is try setting them to different modes before adding them (if you know what mode the others were in when they were added, the same mode would be your best bet),  as they may only return certain data in a certain mode.

Also, the log files should show the actual data being returned when they are not perfectly matching a config, so that will help determine why it is not matching - either a conflicting dp assignment (different types), or a missing required dp (which may not be actually required so making it optional could improve the compatibility).  Or maybe it is just completely different and incompatible with your other fans.

**Reply by Alvanosa-code** [reply]
Hey thanks, I was able to get more results when I put the fan light in a different state, so in the end I could add all 4 fans with the correct device settings.
